src/mm-route.js

```diff
@@ -83,35 +48,58 @@ angular.module('mmRoute', [
   // Arguments:
   //   routes    {Object}    Map of roles to route definitions
   //
-  this.setRoutes = function setRoutes(routes) {
+  this.parseUrls = function parse( urlObj) {
```

Comment:
It's probably a bit cleaner to give the function the same name as the property. The function name will be exposed in stack traces and it it has the same name as the property on `this` it would be easier to track down.